primitives/native-token-management/src/lib.rs

```diff
@@ -119,9 +119,12 @@ mod inherent_provider {
 			C::Api: NativeTokenManagementApi<Block>,
 		{
```

Comment:
Apologies for the stylistic nitpickery but the mid-function `return` bothered me. GH doesn't let me put this in a normal suggestion:
```rust

		pub async fn new<Block, C>(
			client: Arc<C>,
			data_source: &(dyn NativeTokenManagementDataSource + Send + Sync),
			mc_hash: McBlockHash,
			parent_hash: <Block as BlockT>::Hash,
		) -> Result<Self, IDPCreationError>
		where
			Block: BlockT,
			C: HeaderBackend<Block>,
			C: ProvideRuntimeApi<Block> + Send + Sync,
			C::Api: NativeTokenManagementApi<Block>,
		{
			let scripts_opt = client
				.runtime_api()
				.get_main_chain_scripts(parent_hash)
				.map_err(IDPCreationError::GetMainChainScriptsError)?;
			let token_amount = match scripts_opt {
				None => None,
				Some(scripts) => {
					let parent_mc_hash: Option<McBlockHash> =
						get_mc_hash_for_block(client.as_ref(), parent_hash)
							.map_err(IDPCreationError::McHashError)?;
					let token_amount = data_source
						.get_total_native_token_transfer(
							parent_mc_hash,
							mc_hash,
							scripts.native_token_policy_id,
							scripts.native_token_asset_name,
							scripts.illiquid_supply_validator_address,
						)
						.await?;

					Some(token_amount).filter(|amount| amount.0 > 0)
				},
			};
			Ok(Self { token_amount })
		}
```

Reply:
I know this feels wrong coming from haskell, but I'd rather have an early return than nest the code here

Reply:
Sure, I'd just like to point out that single return is not a Haskell thing